src/jit/codegenarmarch.cpp

```diff
@@ -655,8 +655,15 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
                 LclVarDsc* varDsc = &compiler->lvaTable[varNumInp];
 
                 assert(varDsc->lvType == TYP_STRUCT);
-                assert(varDsc->lvOnFrame);   // This struct also must live in the stack frame
-                assert(!varDsc->lvRegister); // And it can't live in a register (SIMD)
+                if (varDsc->lvPromoted)
+                {
+                    NYI_ARM("CodeGen::genPutArgStk - promoted struct");
+                }
+                else
```

Comment:
How about following code to keep ARM64 have different behavior ?
```
if (varDsc->lvPromoted)
{
   NYI_ARM("CodeGen::genPutArgStk - promoted struct");
}
assert(varDsc->lvOnFrame);   // This struct also must live in the stack frame
assert(!varDsc->lvRegister); // And it can't live in a register (SIMD)
```